dixit/src/main/java/tw/wally/dixit/usecases/CreateDixitUseCase.java

```diff
@@ -0,0 +1,82 @@
+package tw.wally.dixit.usecases;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import tw.wally.dixit.model.Dixit;
+import tw.wally.dixit.model.Player;
+import tw.wally.dixit.model.VictoryCondition;
+import tw.wally.dixit.repositories.CardRepository;
+import tw.wally.dixit.repositories.DixitRepository;
+
+import javax.inject.Named;
+import java.util.Collection;
+
+import static tw.wally.dixit.utils.StreamUtils.mapToList;
+
+/**
+ * @author - wally55077@gmail.com
+ */
+@Named
+@AllArgsConstructor
+public class CreateDixitUseCase {
+    private final DixitRepository dixitRepository;
+    private final CardRepository cardRepository;
+
+    public void execute(Request request) {
+        var cards = cardRepository.findAll();
+        var game = request.game;
+        var dixit = new Dixit(game.id, game.gameSetting.toVictoryCondition(), cards);
+        dixit.join(game.host.toPlayer());
+        mapToList(game.players, Gamer::toPlayer).forEach(dixit::join);
+        dixit.start();
+        dixit = dixitRepository.save(dixit);
+        mayPublishEvents(dixit);
```

Comment:
Extract  a `getPlayers` method. Don't mix the DTO conversion code with the use case flow.

```java
        var cards = cardRepository.findAll();
        var dixit = new Dixit(game.id, game.gameSetting.toVictoryCondition(), cards);
        getPlayers(request).forEach(dixit::join);
        dixit.start();
        dixit = dixitRepository.save(dixit);
        mayPublishEvents(dixit);
```

Reply:
@Wally5077 any update?